crates/rune/src/any.rs

```diff
@@ -54,6 +54,51 @@ use crate::hash::Hash;
 ///     Ok(module)
 /// }
 /// ```
+///
+/// ## `#[rune_derive(PROTOCOL, PROTOCOL = handler, ...)]` attribute
+///
+/// Can directly implement supported protocols via the rust Trait, i.e.,
+/// `#[rune_derive(STRING_DEBUG)]` requires the type to implement
+/// [`Debug`](core::fmt::Debug) and adds a handler for
+/// [`Protocol::STRING_DEBUG`](crate::runtime::Protocol::STRING_DEBUG).
+///
+/// For unsupported protocols, or to deviate from the trait implementation,
+/// a custom handler can be specified, this can either be a closure or a path
+/// to a function.
+///
+/// ```
+/// use rune::Any;
+///
+/// #[derive(Any, Debug)]
+/// #[rune_derive(STRING_DEBUG)]
+/// #[rune_derive(INDEX_GET = |it: Self, i: usize| it.0[i])]
+/// struct Struct(Vec<usize>);
+/// ```
+///
+/// ## `#[rune_functions(some_function, ...)]` attribute
+/// 
+/// Allows specifying functions that will be registered together with the type,
+/// these need to be annotated with [`#[rune::function]`](crate::function).
+///
+/// ```
+/// use rune::Any;
+///
+/// #[derive(Any)]
+/// #[rune_functions(Self::first, second)]
+/// struct Struct(bool, usize);
+///
+/// impl Struct {
+///     #[rune::function]
+///     fn first(self) -> bool {
+///         self.0
+///     }
+/// }
+/// 
+/// #[rune::function]
+/// fn second(it: Struct) -> usize {
+///     it.1
+/// }
+/// ```
 pub use rune_macros::Any;
```

Comment:
@udoprog The docs for the `Any` macro are actually not shown. Not sure if that is due to being reexported again at the crate root.

Reply:
AFAICT the only way around that, is moving this reexport in the crate root.

Reply:
https://github.com/rust-lang/rust/issues/81893

Reply:
Welp, time to move!

Reply:
Move the re-export and document it. Otherwise you have a circular dev-dependency for any doc tests that uses rune which rust-analyzer does not like. And all the documentation in one place.